### DISCUSSION TITLE
trying to poulate IP Adresses via python, getting ssl Error

### DISCUSSION THREAD

**Discussion by Kodikodiko** [netbox-community/netbox#14186] [Help Wanted!] [answered]
Hi
I am trying to scan for my devices via a python script and populate the IP addresses.
I am using netbox_python to do so.

relevant code:
NB_URL = "https://192.168.1.102:8681"
netbox = NetBoxClient(base_url="192.168.1.102", token="token goes here")

on request i can paste the whole script here, ofc!

i receive the following error message:
requests.exceptions.SSLError: HTTPSConnectionPool(host='192.168.1.102', port=8681): Max retries exceeded with url: /api/ipam/ip-addresses/?q=192.168.1.1/ (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate (_ssl.c:992)')))

i do have a self signed certificate.

2 options for solving:
somehow correct the connections string, in order that a self signed certificate is accepted
or
set up netbox without SSL at all. This would be okay for me, since i am in a homelab environment

any ideas/hints?

tnx!!!

**Comment by vladimirs-git** [comment, marked as answer]
It appears that netbox_python does not have attributes for SSL verification. 
You can try setting verify=False directly in the Session object.
```python
NB_URL = "https://192.168.1.102:8681/"
netbox = NetBoxClient(base_url="192.168.1.102", token="token goes here")
netbox._session.verify = False
```

**Comment by anaelsilvabh** [comment]
One option is to use the internal IPAM address: http://127.0.0.1:8001.
